log-ballerina/natives.bal

```diff
@@ -103,82 +116,53 @@ public isolated function printInfo(string msg, *KeyValues keyValues, error? 'err
 # + 'error - The error struct to be logged
 public isolated function printWarn(string msg, *KeyValues keyValues, error? 'error = ()) {
-    if (isLogLevelEnabledExtern(WARN)) {
+    if (isLogLevelEnabled(WARN)) {
         print(WARN, msg, keyValues, 'error);
     }
 }
 
 isolated function print(string logLevel, string msg, *KeyValues keyValues, error? err = ()) {
-    string keyValuesString = "";
+    LogRecord logRecord = {
+        time: getCurrentTime(),
+        level: logLevel,
+        module: getModuleName() == "." ? "" : getModuleName(),
+        message: msg
+    };
+    if err is error {
+        logRecord["error"] = err.message();
+    }
     foreach [string, Value] [k, v] in keyValues.entries() {
         anydata value;
         if (v is Valuer) {
             value = v();
         } else {
             value = v;
         }
-        keyValuesString += appendKeyValue(k, value);
+        logRecord[k] = value;
     }
-    printExtern(logLevel, getOutput(msg, keyValuesString, err), format);
-}
-
-isolated function printExtern(string logLevel, string msg, string outputFormat) = @java:Method {
-    'class: "org.ballerinalang.stdlib.log.Utils"
-} external;
-
-isolated function appendKeyValue(string key, anydata value) returns string {
-    string k;
-    string v;
-    if (format == JSON_OUTPUT_FORMAT) {
-        k = string `, "${key}": `;
+    if format == "json" {
+        printJsonExtern(logRecord.toJsonString());
     } else {
-        k = string ` ${key} = `;
+        printLogFmtExtern(logRecord);
     }
-    if (value is string) {
-        v = string `"${value}"`;
-    } else {
-        v = value.toString();
-    }
-    return k + v;
 }
 
-isolated function getOutput(string msg, string keyValues, error? err = ()) returns string {
-    string output = "";
-    if (format == JSON_OUTPUT_FORMAT) {
-        output = string `"message": ${getMessage(msg, err)}${keyValues}`;
-    } else {
-        output = string `message = ${getMessage(msg, err)}${keyValues}`;
-    }
-    return output;
-}
+isolated function printJsonExtern(string msg) = @java:Method {'class: "org.ballerinalang.stdlib.log.Utils"} external;
 
-isolated function getMessage(string msg, error? err = ()) returns string {
-    string message =  string `"${msg}"`;
-    if (err is error) {
-        if (format == JSON_OUTPUT_FORMAT) {
-            message = string `"${msg}", "error": "${err.message()}"`;
-        } else {
-            message = string `"${msg}" error = "${err.message()}"`;
+isolated function printLogFmtExtern(LogRecord logRecord) = @java:Method {'class: "org.ballerinalang.stdlib.log.Utils"} external;
+
+isolated function isLogLevelEnabled(string logLevel) returns boolean {
+    string moduleLogLevel = level;
+    if modules.length() > 0 {
+        string moduleName = getModuleName();
+        if modules.hasKey(moduleName) {
+            var moduleLevel = modules.get(moduleName);
+            moduleLogLevel = moduleLevel.level;
```

Comment:
Can we merge these two lines together

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-log/pull/116/commits/f5917a93ae0cfff80932b9b522c399ce833aa33c